content/blog/2022/2022-08-03-feature-flags-in-nodejs-react-client-with-launchdarkly.md

```diff
@@ -0,0 +1,332 @@
+
+Next we can run the following command to start the UI:
+
+```bash
+npm start
+```
+
+When the user is not set in the LaunchDarkly UI, our React UI looks like below:
+
+{{% image alt="React User List" src="images/posts/nodejs-frontend-launchdarkly/Button.png" %}}
+
+Now we will define “John Doe” as name in LaunchDarkly feature flag variation. Once that is saved, our UI will hide this button. In a real production implementation, this name will be fetched from the logged-in session of the user and matched in Launchdarkly with its ID or username.
+
+## Sorting of Data using Flags
+
+Next we will try to sort our content in the UI using our feature flags. Since our  users are added with some date stamp, we can easily sort our data naturally or based upon the time when that particular user is added in our system. We will create a feature flag variation in LaunchDarkly UI with a boolean type and try to use it in our code.
+
+{{% image alt="Boolean Feature Flag LaunchDarkly" src="images/posts/nodejs-frontend-launchdarkly/User_List_Sorting.png" %}}
+
+Then we will add the sorting logic in our same `App.js`and use it with a switch in the UI. The UI will display either *Natural Sorting* or *Time Sorting*. By default, if the flag is set to true then Time Sorting takes place otherwise Natural Sorting:
+
+```javascript
+import React, { Component } from 'react';
+import './App.css';
+import * as LDClient from 'launchdarkly-js-client-sdk';
+
+const isNewer = (a, b) => Date.parse(a.added) - Date.parse(b.added);
+
+class App extends Component {
+  constructor() {
+    super()
+    this.state = {
+      selectedSortOrder: null,
+      users: [
+        { name: 'John Doe', added: new Date('2022-7-27') },
+        { name: 'Allen Witt', added: new Date('2022-6-30') },
+        { name: 'Cheryl Strong', added: new Date('2022-7-02') },
+        { name: 'Marty Byrde', added: new Date('2022-5-03') },
+        { name: 'Wendy Byrde', added: new Date('2022-6-03') },
+      ]
+    }
+  }
+  componentDidMount() {
+    const user = {
+      // UI based user
+      key: 'user_a'
+    }
+    // SDK requires Client-side ID for UI call
+    this.ldclient = LDClient.initialize('62e*********************', user);
+    this.ldclient.on('ready', this.onLaunchDarklyUpdated.bind(this));
+    this.ldclient.on('change', this.onLaunchDarklyUpdated.bind(this));
+  }
+  onLaunchDarklyUpdated() {
+    this.setState({
+      featureFlags: {
+        defaultSortingIsAdded: this.ldclient.variation('user-list-default-sorting-check', true),
```

Comment:
I have updated the code as well as the article as per your suggestion.

But my free trial has expired and I am not able to update the existing feature flags in LD. I have to refactor rest of the articles as well and I need to create another trial account with a different email id so that it can last for another 15 days. So if you are not planning to publish this article immediately, then I can create that account next week and share the updated screenshots. I have been travelling this week and I won't get much time to update the other articles.